File: src/main/java/com/lakamsani/farefinder/Flight.java
package com.lakamsani.farefinder;

/**
 * Represents a single flight option.
 */
public class Flight {
    private final String airline;
    private final String flightNumber;
    private final int price;
    private final String departureTime;
    private final String arrivalTime;
    private final String duration;

    public Flight(String airline, String flightNumber, int price,
                  String departureTime, String arrivalTime, String duration) {
        this.airline = airline;
        this.flightNumber = flightNumber;
        this.price = price;
        this.departureTime = departureTime;
        this.arrivalTime = arrivalTime;
        this.duration = duration;
    }

    public String getAirline() { return airline; }
    public String getFlightNumber() { return flightNumber; }
    public int getPrice() { return price; }
    public String getDepartureTime() { return departureTime; }
    public String getArrivalTime() { return arrivalTime; }
    public String getDuration() { return duration; }

    @Override
    public String toString() {
        return String.format("Flight{airline='%s', flightNumber='%s', price=%d, " +
                "departureTime='%s', arrivalTime='%s', duration='%s'}",
                airline, flightNumber, price, departureTime, arrivalTime, duration);
    }
}

File: fare_finder/flight.py
from dataclasses import dataclass


@dataclass
class Flight:
    airline: str
    flight_number: str
    price: int
    departure_time: str
    arrival_time: str
    duration: str

    def __str__(self) -> str:
        return (
            f"Flight(airline='{self.airline}', flight_number='{self.flight_number}', "
            f"price={self.price}, departure_time='{self.departure_time}', "
            f"arrival_time='{self.arrival_time}', duration='{self.duration}')"
        )
